### PR TITLE
colima: embed version and git revision

### PR DESCRIPTION
**Pull request by Atemu**
These are shown on `colima version`

See also https://github.com/abiosoft/colima/issues/301

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

cc @aaschmid

### CODE REVIEW

File: pkgs/applications/virtualization/colima/default.nix [hunk 1]
```diff
@@ -14,13 +14,25 @@ buildGoModule rec {
     owner = "abiosoft";
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256-g7q2DmtyArtW7Ii2XF5umXQ0+BlCSa1Q7VNNuIuX65k=";
+    sha256 = "sha256-KYW3gxf21aWnuRHkysOjArzMSNH3m3XDoi6Sic3N+Po=";
+
+    # We need the git revision
+    leaveDotGit = true;
```
Comment (LunNova) [inline pkgs/applications/virtualization/colima/default.nix:20]: Think this is still dangerous to use and the hash will change randomly :(

See #8567

**Review by LunNova — CHANGES_REQUESTED** [review]

**Comment by LunNova** [comment]
For the input-remapper packaged I ended up hardcoding the revision to get around the issues with leaveDotGit.

I'd love to know if there's a better approach.

**Comment by techknowlogick** [comment]
> For the input-remapper packaged I ended up hardcoding the revision to get around the issues with leaveDotGit.

Same pattern used for kubernetes-helm

File: pkgs/applications/virtualization/colima/default.nix (hunk 1 above)
Comment (Atemu) [reply pkgs/applications/virtualization/colima/default.nix:20]: Look closer, I delete the .git dir in the next few lines

**Comment by Atemu** [comment]
Take a look at how I resolved this. If this pattern is needed often, it might be worth integrating into the fetchers.

File: pkgs/applications/virtualization/colima/default.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (Atemu) [reply pkgs/applications/virtualization/colima/default.nix:20]: Please take a look at the next few lines, the .git dir is removed after embedding the revision.

Your code snippet does IFD and that's not allowed in Nixpkgs.

[the remark this answers is not in the document]
Comment (aaschmid) [reply pkgs/applications/virtualization/colima/default.nix:20]: Yes, I saw the `rm` but I am unsure if this solves the indeterminism. We could simply add the ref manually as well but I have no strong opinion on it.

What does IFD mean?

Comment (LunNova) [reply pkgs/applications/virtualization/colima/default.nix:20]: Oops! Looksgood

**Review by LunNova — APPROVED** [review]

File: pkgs/applications/virtualization/colima/default.nix (hunk 1 above)
Comment (LunNova) [reply pkgs/applications/virtualization/colima/default.nix:20]: Hah I feel silly right now. Obviously skimmed this, didn't pay attention, and then ended up proposing a similar change in #8567 which is already what you're doing here. >.>

[the remark this answers is not in the document]
Comment (LunNova) [reply pkgs/applications/virtualization/colima/default.nix:20]: [Import From Derivation](https://nixos.wiki/wiki/Import_From_Derivation).

It's no good for evaluation performance.

**Review by aaschmid — APPROVED** [review]

File: pkgs/applications/virtualization/colima/default.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (Atemu) [reply pkgs/applications/virtualization/colima/default.nix:20]: Also breaks Hydra I believe.

Removing the git dir should fully solve determinism. It doesn't exist anymore, so it's not included in the hash.

**Comment by aaschmid** [comment]
@kalbasit: Can you merge this?

**Comment by aaschmid** [comment]
@Artturin @bobby285271: Can anyone of you merge this? Thanks!

**Pull request merged**